google/ggrc-core
# GGRC-5057 Handle errors on status change

# Issue description

I believe previously the reason of this issue was connected with the case when a user was updating a local custom attribute and was trying to change the status of an assessment. Due to the slow response time from the back-end side the response was returning error. I was not able to reproduce this issue. At the same time I found that the application does not handle back-end errors. In case of changing the status of an assessment the UI part is changing immediately. At the same time in case of getting a server error like 4xx, 5xx, etc (except 409) after that the status will be not returned back and no message will be visible for a user.

# Steps to test the changes

Handle errors (server, network, etc.):

1. Open any assessment.
2. Disable network or ideally is to use a soft for intercepting requests/response and change status codes to 4xx, 5xx.
3. Try to change the status of the assessment. For example by clicking on 'Complete' button.

Expected result: 
1. The status of the assessment was returned to the previous before changing the status.
2. The error message should appear at the left bottom corner.


Resolve conflict:

1. Run two instances of the application. For example using two tabs of a browser.
2. Open the same assessment.
3. Change the assessment's status at one tab.
4. Try to change the assessment's status to another at another tab. 

Expected result:
1. The status should be changed to the status which was applied first.
2. The warning message should appear at the left bottom corner.

# Solution description

Previously in case of changing the status of an assessment only the case with status code 409 was handled. At the same time statuses like 4xx, 5xx were ignored. The solution is to handle all statuses and show message for a user based on defined error messages.

# Sanity checklist


# PR Review checklist
